pkg/controller/gardenletdeployer/actuator.go

```diff
@@ -451,71 +445,25 @@ func (a *Actuator) reconcileSeedSecrets(ctx context.Context, obj client.Object,
-	)
-
 	// Get backup secret
 	backupSecret, originalErr := kubernetesutils.GetSecretByObjectReference(ctx, a.GardenClient, spec.Backup.CredentialsRef)
 
 	if client.IgnoreNotFound(originalErr) != nil {
 		return originalErr
 	}
 
-	if apierrors.IsNotFound(originalErr) && !allowCopying {
-		return fmt.Errorf("the configured backup secret does not exist, however the feature gate DoNotCopyBackupCredentials is enabled and shoot infrastructure credentials will not be reused: %w", originalErr)
-	}
-
-	if originalErr == nil {
-		checksum = utils.ComputeSecretChecksum(backupSecret.Data)[:8]
+	if apierrors.IsNotFound(originalErr) {
+		return fmt.Errorf("the configured backup secret does not exist and shoot infrastructure credentials will not be reused: %w", originalErr)
 	}
 
 	const (
 		secretStatusLabelKey   = "secret.backup.gardener.cloud/status"
 		secretStatusLabelValue = "previously-managed"
 	)
 
-	// Create or update backup secret if it doesn't exist, or is owned by the object, or was previously managed by this controller
-	if allowCopying && (apierrors.IsNotFound(originalErr) || metav1.IsControlledBy(backupSecret, obj) || backupSecret.Labels[secretStatusLabelKey] == secretStatusLabelValue) {
-		gvk, err := apiutil.GVKForObject(obj, a.GardenClient.Scheme())
-		if err != nil {
-			return fmt.Errorf("could not get GroupVersionKind from object %v: %w", obj, err)
-		}
-
-		infrastructureSecret, err := a.GetInfrastructureSecret(ctx)
-		if err != nil {
-			return err
-		}
-
-		// If there is no infrastructure Secret, e.g. WorkloadIdentity is used instead
-		// we skip the copying as it is not supported.
-		if infrastructureSecret != nil {
-			secret := &corev1.Secret{
-				ObjectMeta: metav1.ObjectMeta{Namespace: spec.Backup.CredentialsRef.Namespace, Name: spec.Backup.CredentialsRef.Name},
-			}
-
-			if _, err := controllerutils.CreateOrGetAndStrategicMergePatch(ctx, a.GardenClient, secret, func() error {
-				secret.OwnerReferences = []metav1.OwnerReference{
-					*metav1.NewControllerRef(obj, gvk),
-				}
-				secret.Type = corev1.SecretTypeOpaque
-				secret.Data = infrastructureSecret.Data
-				delete(secret.Labels, secretStatusLabelKey)
-				return nil
-			}); err != nil {
-				return err
-			}
-
-			checksum = utils.ComputeSecretChecksum(secret.Data)[:8]
-		} else if apierrors.IsNotFound(originalErr) {
-			return errors.New("backup is configured to reference a credential, but there is no infrastructure credential to copy")
-		}
-	} else if !allowCopying && metav1.IsControlledBy(backupSecret, obj) {
-		// backup secret was copied at an earlier stage
-		// remove the ownerReference as the controller is no longer responsible for it
+	// If backup secret was copied at an earlier stage, remove the ownerReference as the controller is no longer responsible for it
+	// TODO(dimityrmirchev): Remove this logic when the DoNotCopyBackupCredentials feature gate is removed
```

Comment:
When is expected the feature gate to be removed, I guess after 1.134 is released?

As per https://github.com/gardener/gardener/blob/master/docs/development/process.md#todo-statements


```suggestion
	// TODO(dimityrmirchev): Remove this logic when the DoNotCopyBackupCredentials feature gate is removed, i.e. after v1.134 has been released. 
```